File: 2017-10-DevconDavao/practice-2017-09-30-android/MyApplication/domain/src/main/java/com/example/jboy/interactors/createcat/CreateCatInteractor.java
package com.example.jboy.interactors.createcat;

import com.example.jboy.entities.Cat;
import com.example.jboy.repositories.CatsRepository;

import java.util.UUID;

/**
 * Created by Dev on 30/09/2017.
 */

class CreateCatInteractor {
    private CatsRepository repository;

    CreateCatInteractor(CatsRepository repository) {
        this.repository = repository;
    }

    public CreateCatResponse execute(CreateCatRequest request) {

        Cat cat = new Cat(UUID.randomUUID());
        cat.setName(request.name);
        cat.setTitle(request.title);

        repository.save(cat);

        CreateCatResponse response = new CreateCatResponse();
        response.id = cat.getId();
        response.name = request.name;
        response.title = request.title;

        return response;
    }
}

File: 2017-10-DevconDavao/CatsDemo_InsideOutTDD_DotNetCore/src/Interactors/CreateCatInteractor.cs
﻿using System;
using Domain;
using Domain.Repositories;

namespace Interactors
{
    public class CreateCatInteractor
    {
        private ICatsRepository repository;
        
        public CreateCatInteractor(ICatsRepository repository)
        {
            this.repository = repository;
        }

        public CreateCatResponse Execute(CreateCatRequest request)
        {
            var cat = new Cat
            {
                Name = request.Name
            };

            repository.Save(cat);

            var response = new CreateCatResponse();
            response.Name = cat.Name;
            return response;
        }
    }
}